### PR TITLE
feat: add missing CODEOWNERS and MAINTAINERS file

### PR DESCRIPTION
**Pull request by MDr164**

### CODE REVIEW

**Review by trishankatdatadog — APPROVED** [review]

File: MAINTAINERS [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Maintainers should review changes promptly.
+# Github will auto-assign the below maintainers to a pull request.
+
+"**":
+    - rdimitrov
+    - kommendorkapten
+    - mnm678
+    - znewman01
```
Comment (kommendorkapten) [inline MAINTAINERS:8]: I think @znewman01 left as a maintainer?

Comment (trishankatdatadog) [reply MAINTAINERS:8]: Good point. Zack should say something if that is not the case, but he is still also in the @theupdateframework/go-tuf-maintainers list.

Comment (MDr164) [reply MAINTAINERS:8]: Right, I just copied who was in the maintainers list in the org. If that info was outdated we should booth change it there and in this file.

Comment (kommendorkapten) [reply MAINTAINERS:8]: Yeah that makes sense, we should use that list as authoritative.

**Pull request merged**